### ISSUE TITLE
Many methods on `Parser::Source::Range` from `Offense.highlighted_area` raise `NoMethodError`

### ISSUE THREAD

**Issue by Earlopain** [rubocop/rubocop#12745]
The `highlighted_area` method on offense is public API. However, many of its methods will result in an error because it is not instantiated right: https://github.com/rubocop/rubocop/blob/dea3b768caa6b3b8742516a52f34325110364e76/lib/rubocop/cop/offense.rb#L138-L145

`source_line` will be a string here, while `Parser::Source::Range` expects a `Parser::Source::Buffer`

## Steps to reproduce the problem

```rb
# frozen_string_literal: true

require "rubocop"

source_buffer = Parser::Source::Buffer.new("test.rb")
source_buffer.read
location = Parser::Source::Range.new(source_buffer, 0, 2)

offense = RuboCop::Cop::Offense.new(:warning, location, "unused var", "Lint/UselessAssignment", :corrected)

puts offense.highlighted_area.column
# => .../parser-3.3.0.5/lib/parser/source/range.rb:93:in `column': undefined method `column_for_position' for an instance of String (NoMethodError)
# 
#         @source_buffer.column_for_position(@begin_pos)
#                       ^^^^^^^^^^^^^^^^^^^^
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.61.0 (using Parser 3.3.0.5, rubocop-ast 1.31.1, running on ruby 3.3.0) [x86_64-linux]
  - rubocop-minitest 0.34.5
  - rubocop-rake 0.6.0
  - rubocop-sorbet 0.7.7
```

**Comment by snutij** [comment]
Hi, I had the same conclusion on `Range` initialization few weeks ago: https://github.com/rubocop/rubocop/issues/12527
Feel free to close my issue in the same time if this one get fixed 👍

**Comment by Earlopain** [comment]
Oh, I see. Not sure why I didn't find this while I was searching for duplicates. Thanks for pointing me there instead.
